### PR TITLE
doc: Add section about ostree and bootloaders

### PR DESCRIPTION
**Pull request by jmarrero**

### CODE REVIEW

File: docs/bootloaders.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+nav_order: 11
+---
+
+# Bootloaders
```
Comment (cgwalters) [inline docs/bootloaders.md:5]: I think markdown linters won't like two toplevel `#`; simplest is probably to make everything else `##`

**Review by ericcurtin — APPROVED** [review]
LGTM, I might expand this to speak about Androidy Bootloaders found in ARM Automotive platforms

File: docs/bootloaders.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+## OSTree and grub
+
+For historical reasons, OSTree defaults to detecting the bootloader; if some GRUB files are present then OSTree will default to executing `grub2-mkconfig`.
+
+[Commented out for now, as this can lead to the system not booting in some cases.]: #
+[This can be avoided by setting `sysroot.bootloader=none` (except this should not be set on s390x).]: # 
+
+## GRUB and os-prober
+
+A specific component of GRUB that can significantly impede the reliabilty of OS updates is the `os-prober` aspect, which scans all system block devices.  If one doesn't care about dual booting, avoiding this is a good idea.
```
Comment (ericcurtin) [inline docs/bootloaders.md:31]: s/reliabilty/reliability/g

File: docs/bootloaders.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+## Anaconda
+
+Until very recently the Anaconda project only supported setting up the bootloader (e.g. GRUB) on its own, which requires `grub2-mkconfig` etc.  As of recently, Anaconda now [supports bootupd](https://github.com/rhinstaller/anaconda/pull/5298).
```
Comment (ericcurtin) [inline docs/bootloaders.md:35]: s/Until very recently the Anaconda/Until very recently, the Anaconda/g

File: docs/bootloaders.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+## bootupd
+
+As of recently [the bootupd project](https://github.com/coreos/bootupd/) ships [static grub configs](https://github.com/coreos/bootupd/tree/main/src/grub2) and in this case, the `sysroot.bootloader` should be set to `none` (except on s390x).
```
Comment (ericcurtin) [inline docs/bootloaders.md:39]: s/As of recently/As of recently,/g

**Comment by ericcurtin** [comment]
I found some super minor things, but happy to merge this as is, just say the word

**Comment by jmarrero** [comment]
> I found some super minor things, but happy to merge this as is, just say the word

thanks! and yes please expand this is a needed for auto.

**Comment by ericcurtin** [comment]
I only have merge if everything is green permissions, trying to get this build:

ci/prow/fcos-e2e

from orange to green, it seems stuck or something.

**Comment by cgwalters** [comment]
It doesn't run for docs-only PRs
/override ci/prow/fcos-e2e

**Pull request merged**